src/entities/inputValidator/weighted/inputValidatorWeighted.ts

```diff
@@ -60,4 +72,24 @@ export class InputValidatorWeighted implements InputValidatorBase {
     ): void {
         validateTokensRemoveLiquidity(input, poolState);
     }
+
+    private validateWethIsEth(
+        initPoolInput: InitPoolInputV3,
+        poolState: PoolState,
+    ) {
+        if (!initPoolInput.wethIsEth) {
+            return;
+        }
+        const poolContainsWrappedNativeAsset = !!poolState.tokens.find((a) =>
+            isSameAddress(
+                a.address,
+                NATIVE_ASSETS[initPoolInput.chainId].address,
```

Comment:
Same here with `wrapped`:
```suggestion
                NATIVE_ASSETS[initPoolInput.chainId].wrapped,
```